File: CarePackage/Software/DotNet8DesktopRuntime.cs
﻿namespace CarePackage.Software;

public class DotNet8DesktopRuntime : BaseSoftware
{
    public override string           Key            { get; set; } = "dotnet-8-desktop-runtime";
    public override string           Name           { get; set; } = ".NET 8.0 Desktop Runtime";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Runtime;
    public override string           DownloadName   { get; set; } = "windowsdesktop-runtime-8.0-win-x64.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.dotnet;
    public override string           Homepage       { get; set; } = "https://dot.net";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://download.visualstudio.microsoft.com/download/pr/27bcdd70-ce64-4049-ba24-2b14f9267729/d4a435e55182ce5424a7204c2cf2b3ea/windowsdesktop-runtime-8.0.11-win-x64.exe");
}

File: src/lib/software/variants/dotnet_8_desktop_runtime.py
from src.lib.software import BaseSoftware

class DotNet8DesktopRuntime(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'windowsdesktop-runtime-8.0-win-x64.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://aka.ms/dotnet/8.0/windowsdesktop-runtime-win-x64.exe')
